src/plugins/event-plugins/FetchPlugin.ts

```diff
@@ -143,14 +145,31 @@ export class FetchPlugin extends MonkeyPatched<Window, 'fetch'> {
                 xRayTraceEvent.subsegments[0].http.response = {
                     status: response.status
                 };
+
+                if (is429(response.status)) {
+                    xRayTraceEvent.subsegments[0].throttle = true;
+                    xRayTraceEvent.throttle = true;
+                } else if (is4xx(response.status)) {
+                    xRayTraceEvent.subsegments[0].error = true;
+                    xRayTraceEvent.error = true;
+                } else if (is5xx(response.status)) {
+                    xRayTraceEvent.subsegments[0].fault = true;
+                    xRayTraceEvent.fault = true;
+                }
+
                 const cl = parseInt(response.headers.get('Content-Length'), 10);
                 if (!isNaN(cl)) {
                     xRayTraceEvent.subsegments[0].http.response.content_length = cl;
                 }
             }
 
             if (error) {
-                xRayTraceEvent.subsegments[0].error = true;
```

Comment:
question: are `Error` objects only created for 5xx?

Reply:
If an error is passed in along with a status code of 4xx, it will lead to `xRayTraceEvent.subsegments[0].error = true, xRayTraceEvent.subsegments[0].fault=true, xRayTraceEvent.fault = true, xRayTraceEvent.fault = true`. Is it a case we do not have to worry about?

Reply:
In this branch a network error has occurred. There is no response and by extension no status code.